Frontend/Frontend.Client/Pages/MainPage.razor

```diff
@@ -75,17 +74,29 @@
 
     public void MessageReceived(Message message)
     {
-        messages.Add(message);
+        Message? currentUserMessage = messages.Find(messageInList => message.Hash == messageInList.Hash);
+        if(currentMessage != null)
```

Comment:
currentMessage should be currentUserMessage, but maybe the variable should be MessageToConfirm or something?